Agents/DevInstAgent/DevInstAgent/src/test/java/uk/ac/cam/cares/jps/agent/devinst/DevInstQureyBuilderTest.java

```diff
@@ -0,0 +1,352 @@
+
+        //SmartSensor
+        query = Queries.SELECT();
+        query.where(query.var().isA(SmartSensor)).prefix(P_DEV);
+        result = storeClient.executeQuery(query.getQueryString());
+        Assert.assertTrue(result.length() == 1);
+        obtainedIRI = result.getJSONObject(0).getString("x0");
+        Assert.assertTrue(obtainedIRI.contains("ProximitySensor_FH-02"));
+
+
+        //Sensor
+        query = Queries.SELECT();
+        query.where(query.var().isA(Sensor)).prefix(P_DEV, P_SAREF);
+        result = storeClient.executeQuery(query.getQueryString());
+        Assert.assertTrue(result.length() == 1);
+        obtainedIRI = result.getJSONObject(0).getString("x0");
+        Assert.assertTrue(obtainedIRI.contains("HCSR04") && !obtainedIRI.contains("HCSR04_ProximitySensor"));
+
+        //Proximity Sensor
+        query = Queries.SELECT();
+        query.where(query.var().isA(ProximitySensor)).prefix(P_DEV);
+        result = storeClient.executeQuery(query.getQueryString());
+        Assert.assertTrue(result.length() == 1);
+        obtainedIRI = result.getJSONObject(0).getString("x0");
+        Assert.assertTrue(obtainedIRI.contains("HCSR04_ProximitySensor"));
+
+        //Raw Readings
+        query = Queries.SELECT();
+        query.where(query.var().has(measures, query.var())).prefix(P_DEV);
+        result = storeClient.executeQuery(query.getQueryString());
+        Assert.assertTrue(result.length() == 1);
+        obtainedIRI = result.getJSONObject(0).getString("x1");
+        Assert.assertTrue(obtainedIRI.contains("AvgDist_FH02"));
+
+        //Task
+        query = Queries.SELECT();
+        query.where(query.var().has(accomplishes, query.var()));
+        result = storeClient.executeQuery(query.getQueryString());
+        Assert.assertTrue(result.length() == 3);
+
+    }
+
+    @Ignore("Test containers requires docker to function")
+    @Test
+    public void testFindIRI(){
+        //Add dummyraw var here
+        ModifyQuery modify = Queries.MODIFY();
+        String exampleRawString = "http://www.example.com/prefix/api_AvgDist_FH02";
+        Iri exampleRaw = iri(exampleRawString);
+        Iri Length = iri("http://www.ontology-of-units-of-measure.org/resource/om-2/Length");
+        modify.insert(exampleRaw.isA(Length));
+        storeClient.executeUpdate(modify.getQueryString());
+
+        //Check if IRI is consistent
+        queryBuilder.InsertDevice(exampleFind);
+
+        SelectQuery query = Queries.SELECT();
+        query.where(query.var().has(measures, query.var())).prefix(P_DEV);
+        JSONArray result = storeClient.executeQuery(query.getQueryString());
+        Assert.assertTrue(result.length() == 1);
+        String obtainedIRI = result.getJSONObject(0).getString("x1");
+        //System.out.println(obtainedIRI);
+        Assert.assertEquals(obtainedIRI, exampleRawString);
+    }
+
+    @Ignore("Test containers requires docker to function")
+    @Test
+    public void testAdditionalQuery() {
```

Comment:
certain tests seems to fail when runned: testInstantiation(), testFindIRI() and testAdditionalQuery(), please check and modify accordingly